drivers/serial/uart_native_tty.c

```diff
@@ -195,6 +217,9 @@ static int native_tty_serial_init(const struct device *dev)
 		ERROR("%s: could not configure serial port %s\n", dev->name, data->serial_port);
 	}
 
+	struct uart_config *uart_cfg = ((struct native_tty_data *)dev->data)->uart_config;
```

Comment:
Wouldnt it be enough to do it once at the beginning of `native_tty_configure()`?

Reply:
right.. I see. And then `uart_cfg` is blindly set every time and its validity is determined by the error code of  the `configure` function.